#### About

> Effect size

Effect size is a measure of the magnitude of the difference between two groups or populations. It is an important concept in inferential statistics as it helps to understand the practical significance of statistical results.

There are several ways to calculate effect size for different types of data. One commonly used measure is Cohen's d, which is defined as the difference between the means of two groups divided by the pooled standard deviation. 



In [1]:
import numpy as np
from scipy import stats

In [2]:
group1 = np.array([1, 2, 3, 4, 5])
group2 = np.array([6, 7, 8, 9, 10])

In [3]:
mean1 = np.mean(group1)
mean2 = np.mean(group2)
std1 = np.std(group1, ddof=1)
std2 = np.std(group2, ddof=1)


In [4]:
# Calculate the pooled standard deviation
pooled_std = np.sqrt(((len(group1) - 1) * std1 ** 2 + (len(group2) - 1) * std2 ** 2) / (len(group1) + len(group2) - 2))


In [5]:
# Calculate Cohen's d
cohens_d = (mean1 - mean2) / pooled_std

In [6]:
print("Cohen's d:", cohens_d)

Cohen's d: -3.162277660168379


The value of Cohen's d as -3.16, indicating a large effect size. The negative sign indicates that group 2 has a higher mean than group 1, and the absolute value of d indicates the magnitude of the difference relative to the variability within the groups.

